pandas/_libs/algos.pyx

```diff
@@ -1403,99 +1406,62 @@ def rank_2d(
     nans_rank_highest = ascending ^ (na_option == 'top')
     if check_mask:
         nan_fill_val = get_rank_nan_fill_val[rank_t](nans_rank_highest)
+
         if rank_t is object:
-            mask = missing.isnaobj2d(values)
+            mask = missing.isnaobj2d(values).view(np.uint8)
         elif rank_t is float64_t:
-            mask = np.isnan(values)
+            mask = np.isnan(values).view(np.uint8)
```

Comment:
are these views necessary?  or just being extra-explicit?

Reply:
I don't see why they'd be necessary (tests pass both ways), just added to make consistent with existing `rank_1d` pattern that does this.

Reply:
thanks.  no objection, just curious